app/controllers/assignments_controller.rb

```diff
@@ -104,15 +104,7 @@ def student_interface
     if @grouping.nil?
       if @assignment.group_max == 1 && !@assignment.scanned_exam
         begin
-          # fix for issue #627
-          # currently create_group_for_working_alone_student only returns false
-          # when saving a grouping throws an exception
-          unless @student.create_group_for_working_alone_student(@assignment.id)
-            # if create_group_for_working_alone_student returned false then the student
-            # must have an ( empty ) existing grouping that he is not a member of.
-            # we must delete this grouping for the transaction to succeed.
-            Grouping.find_by_group_id_and_assignment_id( Group.find_by_group_name(@student.user_name), @assignment.id).destroy
-          end
+          @student.create_group_for_working_alone_student(@assignment.id)
         rescue RuntimeError => @error
           flash_message(:error, 'Error')
```

Comment:
Do you mind changing this to use the error's message? (And I think you can change `@error` to `error`). I know it's not exactly relevant, but I think it's a related-enough improvement.